File: framework/security/src/main/java/top/bettercode/simpleframework/security/AdditionalUserDetails.java
package top.bettercode.simpleframework.security;

import java.util.Collection;
import java.util.HashMap;
import java.util.Map;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.User;

public class AdditionalUserDetails extends User {

  private static final long serialVersionUID = 1L;

  private final Map<String, Object> additionalInformation = new HashMap<>();

  public AdditionalUserDetails(String username, String password,
      Collection<? extends GrantedAuthority> authorities) {
    super(username, password, authorities);
  }

  public AdditionalUserDetails(String username, String password, boolean enabled,
      boolean accountNonExpired,
      boolean credentialsNonExpired, boolean accountNonLocked,
      Collection<? extends GrantedAuthority> authorities) {
    super(username, password, enabled, accountNonExpired, credentialsNonExpired, accountNonLocked,
        authorities);
  }

  public void put(String key, Object value) {
    this.additionalInformation.put(key, value);
  }

  public Object get(String key){
    return this.additionalInformation.get(key);
  }

  public Map<String, Object> getAdditionalInformation() {
    return additionalInformation;
  }
}

File: security/src/main/kotlin/top/bettercode/summer/security/userdetails/AdditionalUserDetails.kt
package top.bettercode.summer.security.userdetails

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.User

open class AdditionalUserDetails : User {

    val additionalInformation: MutableMap<String, Any?> = HashMap()

    constructor(
            username: String?, password: String?,
            authorities: Collection<GrantedAuthority?>?
    ) : super(username, password, authorities)

    constructor(
            username: String?, password: String?, enabled: Boolean,
            accountNonExpired: Boolean,
            credentialsNonExpired: Boolean, accountNonLocked: Boolean,
            authorities: Collection<GrantedAuthority?>?
    ) : super(username, password, enabled, accountNonExpired, credentialsNonExpired, accountNonLocked,
            authorities)

    fun put(key: String, value: Any?) {
        additionalInformation[key] = value
    }

    operator fun get(key: String): Any? {
        return additionalInformation[key]
    }

}
